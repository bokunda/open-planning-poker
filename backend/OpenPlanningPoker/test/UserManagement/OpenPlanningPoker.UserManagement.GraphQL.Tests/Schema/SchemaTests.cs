namespace OpenPlanningPoker.UserManagement.GraphQL.Tests.Schema;


public class SchemaTests
{
    [Fact]
    public async Task GenerateSchema()
    {
        // arrange
        var schema = await new ServiceCollection()
            .AddGraphQL()
            .AddAuthorization()
            .AddTypes()
            .BuildSchemaAsync();

        // act
        var schemaSDL = schema.ToString();

        // assert
        schemaSDL.MatchSnapshot();
    }
    
}
