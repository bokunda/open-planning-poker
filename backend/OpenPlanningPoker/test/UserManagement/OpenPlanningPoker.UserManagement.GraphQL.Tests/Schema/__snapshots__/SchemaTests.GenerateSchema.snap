schema {
  query: Query
  mutation: Mutation
}

type Mutation {
  ping: String!
  registerUser(username: String): User!
}

type Query {
  ping: String!
}

type User {
  id: UUID!
  username: String!
}

"The `@specifiedBy` directive is used within the type system definition language to provide a URL for specifying the behavior of custom scalar definitions."
directive @specifiedBy("The specifiedBy URL points to a human-readable specification. This field will only read a result for scalar types." url: String!) on SCALAR

scalar UUID @specifiedBy(url: "https:\/\/tools.ietf.org\/html\/rfc4122")
